databricks/koalas/series.py

```diff
@@ -1210,8 +1227,72 @@ def fillna(self, value=None, axis=None, inplace=False):
         4    0.0
         5    6.0
         Name: x, dtype: float64
+
+        We can also propagate non-null values forward or backward.
+
+        >>> s.fillna(method='ffill')
+        0    NaN
+        1    2.0
+        2    3.0
+        3    4.0
+        4    4.0
+        5    6.0
+        Name: x, dtype: float64
+
+        >>> s = ks.Series([np.nan, 'a', 'b', 'c', np.nan], name='x')
+        >>> s.fillna(method='ffill')
+        0    None
+        1       a
+        2       b
+        3       c
+        4       c
+        Name: x, dtype: object
         """
-        kseries = _col(self.to_dataframe().fillna(value=value, axis=axis, inplace=False))
+        if self.isnull().sum() == 0:
+            if inplace:
+                return
+            else:
+                return self
```

Comment:
Maybe we should return a copy to avoid the effect of the following inplace-like operations?